src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -801,6 +801,13 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
         {
             var projectInstalledPackageReferences = await nuGetProject.GetInstalledPackagesAsync(token);
 
+            NuGetFramework framework;
+            if (!nuGetProject.TryGetMetadata<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework, out framework))
+            {
+                // Default to the any framework if the project does not specify a framework.
```

Comment:
This is the existing code, nothing new Look at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L3208 

And I'd imagine this would have been added as fallback to proceed since this will not make any real impact difference because these are temp PR actions. It will then run `PreviewBuildIntegratedProjectActionsAsync` to do preview restore which will use the correct information from PackageSpec